src/strands/experimental/bidirectional_streaming/types/bidirectional_streaming.py

```diff
@@ -6,231 +6,483 @@
+                "image": image,
+                "mime_type": mime_type,
+            }
+        )
 
-class TextOutputEvent(TypedDict):
-    """Text output event from the model during bidirectional streaming.
+    @property
+    def image(self) -> str:
+        return cast(str, self.get("image"))
 
-    Attributes:
-        text: The text content from the model.
-        role: The role of the message sender.
-    """
+    @property
+    def mime_type(self) -> str:
+        return cast(str, self.get("mime_type"))
 
-    text: str
-    role: Role
 
+# ============================================================================
+# Output Events (received via session.receive_events())
+# ============================================================================
 
-class TranscriptEvent(TypedDict):
-    """Transcript event for audio transcriptions.
-    
-    Used for both input transcriptions (user speech) and output transcriptions
-    (model audio). These are informational and separate from actual text responses.
-    
-    Attributes:
-        text: The transcribed text.
-        role: The role of the speaker ("user" or "assistant").
-        type: Type of transcription ("input" or "output").
+
+class SessionStartEvent(TypedEvent):
+    """Session established and ready for interaction.
+
+    Parameters:
+        session_id: Unique identifier for this session.
+        model: Model identifier (e.g., "gpt-realtime", "gemini-2.0-flash-live").
+        capabilities: List of supported features (e.g., ["audio", "tools", "images"]).
     """
-    
-    text: str
-    role: Role
-    type: Literal["input", "output"]
 
+    def __init__(self, session_id: str, model: str, capabilities: List[str]):
+        super().__init__(
+            {
+                "type": "bidirectional_session_start",
+                "session_id": session_id,
+                "model": model,
+                "capabilities": capabilities,
+            }
+        )
 
-class InterruptionDetectedEvent(TypedDict):
-    """Interruption detection event.
+    @property
+    def session_id(self) -> str:
+        return cast(str, self.get("session_id"))
 
-    Signals when user interruption is detected during model generation.
+    @property
+    def model(self) -> str:
+        return cast(str, self.get("model"))
 
-    Attributes:
-        reason: Interruption reason from predefined set.
+    @property
+    def capabilities(self) -> List[str]:
+        return cast(List[str], self.get("capabilities"))
+
+
+class TurnStartEvent(TypedEvent):
+    """Model starts generating a response.
+
+    Parameters:
+        turn_id: Unique identifier for this turn (used in turn.complete).
     """
 
-    reason: Literal["user_input", "vad_detected", "manual"]
+    def __init__(self, turn_id: str):
+        super().__init__({"type": "bidirectional_turn_start", "turn_id": turn_id})
 
+    @property
+    def turn_id(self) -> str:
+        return cast(str, self.get("turn_id"))
```

Comment:
Don't want to over-index, but this is the first time we're talking about `turn`s - did we want to consider `invocation`? 

(I don't think so, just asking)